shared/common-adapters/copy-text.tsx

```diff
@@ -35,6 +35,9 @@ const CopyText = (props: Props) => {
   const dispatch = Container.useDispatch()
   const copyToClipboard = (text: string) => dispatch(ConfigGen.createCopyToClipboard({text}))
   const copy = () => {
+    if (props.withReveal && !revealed) {
+      reveal()
+    }
```

Comment:
Is this intentional? It seems copying shouldn't reveal the content in UI.

Reply:
Yeah, it's intentional - I was specifically thinking of the copyText usage for stellar secret keys, which doesn't actually decrypt the secret key until `onReveal()` is called. (although that particular use case is still a little wonky here because `copyToClipboard` doesn't wait for whatever `onReveal` is doing)

Reply:
I see, interesting. It seems the current interface is a bit confusing. Here's some ideas: perhaps we could add these props to the `CopyText` component:

```
loadText?: ()=>void
textPlaceholder?: string
```

If `text` is empty, we show `textPlaceholder` when revealed. Otherwise, show text when revealed. If `withReveal` is not set, we'd just call `loadText` in a `React.Effect` with empty dependent (i.e. like `componentDidMount`). For copy, we'd need to add a state (e.g. `requestedCopy`) to track if user has requested copy, and in the `onClick` event for the copy button we simply flip `requestedCopy`, and call `loadText()`. Then we need an effect that does the actual copy thing when `text` becomes non-empty, _and_ flip `requestedCopy` back.

In wallet's setting here https://github.com/keybase/client/blob/55f3b947934aff4b70c2c472466ca90ab07a4d23/shared/wallets/wallet/settings/index.tsx#L147 , we could just pass in `secretKey` as text, and that string as `textPlaceholder`.

Sorry it's a lot larger than it seems. What do you think? I'm open to other ideas!

Reply:
Sounds like a good plan to me! I'll implement that.